layer/mounted_layer.go

```diff
@@ -50,11 +53,68 @@ func (ml *mountedLayer) Parent() Layer {
 	return nil
 }
 
+func (ml *mountedLayer) incActiveMount() {
+	ml.activeMountL.Lock()
+	ml.activeMount++
+	ml.activeMountL.Unlock()
+}
+
+func (ml *mountedLayer) decActiveMount() {
+	ml.activeMountL.Lock()
+	ml.activeMount--
+	ml.activeMountL.Unlock()
+}
+
 func (ml *mountedLayer) Mount(mountLabel string) (string, error) {
-	return ml.layerStore.driver.Get(ml.mountID, mountLabel)
+	var path string
+	var err error
+
+	ml.activeMountL.Lock()
+	activeMount := ml.activeMount
+	ml.activeMountL.Unlock()
+
+	if activeMount > 0 {
+		path = ml.pathCache // load from cache.
+		if path == "" {
+			path, err = ml.layerStore.store.GetMountPath(ml.name) // load from disk
+			if err != nil {
+				logrus.Debugf("mountedLayer GetMountPath failed with err %s", err)
+				return "", err
+			}
+		}
+		ml.incActiveMount()
+		return path, nil
+	}
+
+	path, err = ml.layerStore.driver.Get(ml.mountID, mountLabel)
+	if err != nil {
+		logrus.Debugf("mountedLayer Mount Get failed due to err: %s", err)
+		return "", err
+	}
+
+	if ml.pathCache == "" {
```

Comment:
Why this check? If it was an actual mount then I think it's safer to save again. Graphdrivers don't need to guarantee the consistency of this path.